core/util/searx.py

```diff
@@ -0,0 +1,107 @@
+"""
+OWASP Maryam!
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+class main:
+
+	def __init__(self, q, limit=1):
+		""" searx search engine
+
+			q	: Query for search
+			limit	: Number of Pages
+
+		"""
+		self.framework = main.framework
+		self.q = q
+		self._pages = ''
+		self.limit = limit
+		self._links = []
+		self._json = {}
+		self.searx = ['https://searx.prvcy.eu/search', 'https://searx.xyz/search', 'https://searx.bar/search']
+
+	def run_crawl(self):
+		self._json['results'] = [] 
+		for url in self.searx:
+			params = {'category_general': 1, 'q': self.q, 
+				  'pageno': 1, 'time_range': None, 
+				  'format': 'json'}
+			max_attempts = 0
+			while True:
+				self.framework.verbose(f"[SEARX] Searching {url} page {params['pageno']}...", end='\r')
+				try:
+					req = self.framework.request(
+						url=url,
+						params=params,
+						allow_redirects= True)
+				except Exception as e:
+					self.framework.error(f"ConnectionError: {e}", 'util/searx', 'run_crawl')
+					max_attempts += 1
+					if max_attempts == self.limit:
+						self.framework.error(f"Searx {url} is missed!", 'util/searx', 'run_crawl')
+						return
+				else:
+					self._pages += req.text
+					try:
```

Comment:
> if the url hasn't returned anything, means the site was down. so, we should research the query along with the same page to another url/engine.

if the result is not found on one URL then it will anyway search the same query on other URL

> also, don't use try/except for such things

ok

Reply:
if the result is not found on one URL then it will anyway search the same query on other URL? good. question. no problem.

Reply:
if the site is down then it will show the error message because of the except statement on line 49

Reply:
Okay. if there is an error, use another engine.